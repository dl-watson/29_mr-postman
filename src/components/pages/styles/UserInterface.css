.UserInterface {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    font-family: system-ui;

    .formElement {
        display: flex;
        align-content: center;
        justify-content: center;
        flex-direction: column;
    }

    .formElement, .URLInput, .JSONInput {
        display: flex;
        justify-content: center;
        align-items: center;
    }

    .formElement > label {
        margin-left: 20px;
        padding-bottom: 20px;
    }

    .formElement > label > input {
        margin: 5px;
    }

    .URLInput {
        padding: 20px;
    }

    .URLInput > input, button {
        border-radius: 12px;
        height: 30px;
    }

    .URLInput > input {
        width: 80vw;
    }

    .URLInput > button {
        width: 20vh;
    }

    .buttonContainer {
        padding-bottom: 20px;
    }

    .buttonContainer > label {
        margin-left: 20px;
    }

    .textArea {
        display: block;
        width: inherit;
        resize: both;
        overflow: visible;
        min-height: 40px;
        line-height: 20px;
        border: 1px solid #ccc;
        padding: 1px 6px;
        font-family: inherit;
        font-size: inherit;
    }

}
