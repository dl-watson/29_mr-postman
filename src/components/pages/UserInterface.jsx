import React from "react";
import PropTypes from "prop-types";
import TextAreaAutosize from "react-textarea-autosize";
import styles from "./styles/UserInterface.css";

const UserInterface = ({ handleChange, handleSubmit, url, method }) => {
  return (
    <main className={styles.UserInterface}>
      <form className={styles.formElement} onSubmit={handleSubmit}>
        <div className={styles.URLInput}>
          <input
            placeholder="URL to query ..."
            name="url"
            value={url}
            onChange={handleChange}
          />
          <button type="submit">Go</button>
        </div>
        <div className={styles.buttonContainer}>
          <label htmlFor="GET">
            GET
            <input
              name="method"
              id="GET"
              value="GET"
              type="radio"
              checked={method === "GET"}
              onChange={handleChange}
            />
          </label>
          <label htmlFor="PUT">
            PUT
            <input
              name="method"
              id="PUT"
              value="PUT"
              type="radio"
              checked={method === "PUT"}
              onChange={handleChange}
            />
          </label>
          <label htmlFor="POST">
            POST
            <input
              name="method"
              id="POST"
              value="POST"
              type="radio"
              checked={method === "POST"}
              onChange={handleChange}
            />
          </label>
          <label htmlFor="DELETE">
            DELETE
            <input
              name="method"
              id="DELETE"
              value="DELETE"
              type="radio"
              checked={method === "DELETE"}
              onChange={handleChange}
            />
          </label>
        </div>
        <div className={styles.JSONInput}>
          <TextAreaAutosize
            className={styles.textArea}
            name="json"
            onChange={handleChange}
          />
        </div>
      </form>
    </main>
  );
};

UserInterface.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  url: PropTypes.string,
  method: PropTypes.string,
};

export default UserInterface;
