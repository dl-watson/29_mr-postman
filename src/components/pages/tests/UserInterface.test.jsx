import React from "react";
import { render, cleanup } from "@testing-library/react";
import UserInterface from "../UserInterface";

describe("UserInterface component", () => {
  afterEach(() => cleanup());
  const testFunction = () => true;
  it("renders UserInterface", () => {
    const { asFragment } = render(
      <UserInterface
        handleChange={testFunction}
        handleSubmit={testFunction}
        results={"results"}
        method={"method"}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
