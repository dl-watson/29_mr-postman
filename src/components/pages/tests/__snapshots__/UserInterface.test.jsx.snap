// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInterface component renders UserInterface 1`] = `
<DocumentFragment>
  <main
    class="UserInterface"
  >
    <form
      class="formElement"
    >
      <div
        class="URLInput"
      >
        <input
          name="url"
          placeholder="URL to query ..."
          value=""
        />
        <button
          type="submit"
        >
          Go
        </button>
      </div>
      <div
        class="buttonContainer"
      >
        <label
          for="GET"
        >
          GET
          <input
            id="GET"
            name="method"
            type="radio"
            value="GET"
          />
        </label>
        <label
          for="PUT"
        >
          PUT
          <input
            id="PUT"
            name="method"
            type="radio"
            value="PUT"
          />
        </label>
        <label
          for="POST"
        >
          POST
          <input
            id="POST"
            name="method"
            type="radio"
            value="POST"
          />
        </label>
        <label
          for="DELETE"
        >
          DELETE
          <input
            id="DELETE"
            name="method"
            type="radio"
            value="DELETE"
          />
        </label>
      </div>
      <div
        class="JSONInput"
      >
        <textarea
          class="textArea"
          name="json"
        />
      </div>
    </form>
  </main>
</DocumentFragment>
`;
